src/SourceFile.js

```diff
@@ -119,91 +148,82 @@ class SourceFile {
      * The content of the file, stored as regular Javascript string
      * encoded in UTF-8.
      *
-     * @type {String}
+     * @type {String|undefined}
      * @protected
      */
     content;
-    
+
     /**
      * Get the content of this file encoded as a regular Javascript
      * string.
-     * @returns {String} the content of the file, encoded as a JS string
+     * @returns {String|undefined} the content of the file, encoded as a JS string
      */
     getContent() {
         if (this.content === undefined) {
             this.read();
         }
         return this.content;
     }
-    
+
     /**
      * Get the content of this file, encoded as an array of lines. Each
      * line has its trailing newline character removed.
-     * @returns {Array.<String>} the content as an array of lines
+     * @returns {Array.<String>|undefined} the content as an array of lines
      */
     getLines() {
-        if (this.content === undefined) {
+        if (typeof(this.content) === 'undefined') {
```

Comment:
why?

Reply:
content = "" is valid, but `this.content === undefined` returns true, or at least it does on node v20.10.0

Reply:
Hold up, you're saying that after assigning empty string to this.content, it passes strict equality check with undefined?